api/v1beta1/gcpmachine_types.go

```diff
@@ -66,6 +66,62 @@ const (
 	IPForwardingDisabled IPForwarding = "Disabled"
 )
 
+type SecureBootPolicy string
+
+const (
+	// SecureBootPolicyEnable enables the secure boot configuration for the GCP machine.
+	SecureBootPolicyEnable SecureBootPolicy = "Enabled"
+	// SecureBootPolicyDisable disables the secure boot configuration for the GCP machine.
+	SecureBootPolicyDisable SecureBootPolicy = "Disable"
+)
+
+type VTPMPolicy string
+
+const (
+	// VTPMPolicyEnable enables the virtualized trusted platform module configuration for the GCP machine.
+	VTPMPolicyEnable VTPMPolicy = "Enabled"
+	// VTPMPolicyDisable disables the virtualized trusted platform module configuration for the GCP machine.
+	VTPMPolicyDisable VTPMPolicy = "Disable"
+)
+
+type IntegrityMonitoringPolicy string
+
+const (
+	// IntegrityMonitoringPolicyEnable enables integrity monitoring for the GCP machine.
+	IntegrityMonitoringPolicyEnable IntegrityMonitoringPolicy = "Enabled"
+	// IntegrityMonitoringPolicyDisable disables integrity monitoring for the GCP machine.
+	IntegrityMonitoringPolicyDisable IntegrityMonitoringPolicy = "Disable"
+)
+
+// GCPShieldedInstanceConfig describes the shielded VM configuration of the instance on GCP.
+// Shielded VM configuration allow users to enable and disable Secure Boot, vTPM, and Integrity Monitoring
+type GCPShieldedInstanceConfig struct {
+	// SecureBoot Defines whether the instance should have secure boot enabled.
+	// Secure Boot verify the digital signature of all boot components, and halting the boot process if signature verification fails.
+	// If omitted, the platform chooses a default, which is subject to change over time, currently that default is Disabled.
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +kubebuilder:default=Disabled
+	//+optional
+	SecureBoot SecureBootPolicy `json:"secureBoot,omitempty"`
+
+	// VTPM enable virtualized trusted platform module measurements to create a known good boot integrity policy baseline.
+	// The integrity policy baseline is used for comparison with measurements from subsequent VM boots to determine if anything has changed.
+	// If omitted, the platform chooses a default, which is subject to change over time, currently that default is Enabled.
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +kubebuilder:default=Enabled
+	// +optional
+	VTPM VTPMPolicy `json:"vTPM,omitempty"`
```

Comment:
Acronyms aren't particularly user friendly, do we want to spell this out as something more explicit?

Reply:
Well, it's called `EnableVtpm` in the compute api.